csharp/src/Ice/Connection.cs

```diff
@@ -87,9 +49,45 @@ public Acm Acm
         /// <value>The endpoint from which the connection was created.</value>
         public Endpoint Endpoint { get; }
 
+        /// <summary>Gets the connection idle timeout.</summary>
+        public TimeSpan IdleTimeout
+        {
+            get
+            {
+                lock (_mutex)
+                {
+                    return Transceiver.IdleTimeout;
+                }
+            }
+            set
+            {
+                lock (_mutex)
+                {
+                    if (_state == ConnectionState.Active)
+                    {
+                        // Setting the IdleTimeout might throw if it's not supported by the underlying transport. For
+                        // example with Slic, the idle timeout is negotiated when the connection is established, it
+                        // can't be updated after.
+                        Transceiver.IdleTimeout = value;
+
+                        _timer?.Dispose();
+                        TimeSpan period = value / 2;
+                        if (period != TimeSpan.Zero)
+                        {
+                            _timer = new Timer(value => Monitor(), null, period, period);
+                        }
+                    }
+                }
+            }
+        }
+
         /// <summary><c>true</c> for incoming connections <c>false</c> otherwise.</summary>
         public bool IsIncoming => _connector == null;
 
+        /// <summary>Enables or disables the keep alive. When enabled, the connection is kept alive by sending ping
+        /// frames at regular time intervals when the connection is idle.</summary>
+        public bool KeepAlive { get; set; }
```

Comment:
In many use cases keep alive should be disabled to allow idle connections to be closed so being able to disable/enable keep alive is useful. 

For applications that want to maintain the connection alive (typically because they associate state with the connection, such as a session), it can easily be decided at design time whose responsible for enabling the keep alive (the client or the server) so I'm not convinced we need to add negotiation to optimise the use case where both side of the connections would enable this.

Also, given that it's a setting which would be updated after connection establishment, negotiating this setting between the two peers would require an extra Ice2 frame to allow re-negotiating settings that have been established at connection establishment time with the Initialize frame (unless we make this settings a global communicator-level setting only).

I'm going to merge this branch without changing this but let me know if you think we should handle this differently.